### PR TITLE
Add a proof of the extension lemma (needed for the universal property of Patch)

### PR DESCRIPTION
**Pull request by ayberkt**

### CODE REVIEW

**Comment by ayberkt** [comment]
@martinescardo The proof of the extension lemma is now complete!

In addition to this proof, this PR also implements the change to the definition of zero-dimensionality we discussed (which I had also mentioned in #95). The new definition of zero-dimensionality requires closure of covering families under finite joins.

**Review by martinescardo — CHANGES_REQUESTED** [review]

File: source/Locales/BooleanAlgebra.lagda [hunk 1]
```diff
@@ -0,0 +1,719 @@
...
+\begin{code}
+
+_is-sublattice-of_ : BooleanAlgebra 𝓤′ 𝓥′ → Frame 𝓤 𝓥 𝓦 → Ω (𝓤′ ⊔ 𝓤)
+_is-sublattice-of_ B L = Ǝ η ∶ (⟪ B ⟫ → ⟨ L ⟩) , is-embedding B L η holds
+
+\end{code}
+
+\begin{code}
+
+embedding-is-order-isomorphism : (B : BooleanAlgebra 𝓤′ 𝓥′) (L : Frame 𝓤 𝓥 𝓦)
```
Comment (martinescardo) [inline source/Locales/BooleanAlgebra.lagda:250]: This is not what an order isomorphism is. What you have here is "order preserving and reflecting".

Comment (martinescardo) [reply source/Locales/BooleanAlgebra.lagda:250]: So rename the function, please.

File: source/Locales/BooleanAlgebra.lagda [hunk 2]
```diff
@@ -0,0 +1,719 @@
...
+                → (η : ⟪ B ⟫ → ⟨ L ⟩)
+                → is-embedding B L η holds
+                → is-spectral L holds
+                → is-spectral L′ holds
+                → is-spectral′ B L η holds
+                → is-generated-by L B η holds
+                → contains-compact-opens L B η holds
+                → (h : ⟪ B ⟫ → ⟨ L′ ⟩)
+                → is-lattice-homomorphism B L′ h holds
+                → is-contr
```
Comment (martinescardo) [inline source/Locales/BooleanAlgebra.lagda:333]: `is-contr (\Sigma ...)` can be abbreviated as `\exists! ...`.

**Review by martinescardo — CHANGES_REQUESTED** [review]

File: source/Locales/PatchProperties.lagda [hunk 3]
```diff
@@ -697,6 +697,9 @@ We define the following basis for Patch:
         (𝔠 k ∧[ 𝒪 Patch-X ] 𝔬 l) holds
    ※ = ∧-complement (𝒪 Patch-X) † ‡
 
+ ℬ-patch-↑ : Fam 𝓤 ⟨ 𝒪 Patchₛ-X ⟩
```
[the remark this answers is not in the document]
Comment (ayberkt) [reply source/Locales/PatchProperties.lagda:700]: Done.

File: source/Locales/BooleanAlgebra.lagda (hunk 1 above)
Comment (ayberkt) [reply source/Locales/BooleanAlgebra.lagda]: Renamed to `embedding-preserves-and-reflects-order`.

File: source/Locales/BooleanAlgebra.lagda (hunk 2 above)
Comment (ayberkt) [reply source/Locales/BooleanAlgebra.lagda]: Done.

**Comment by ayberkt** [comment]
@martinescardo Should be ready to merge now.

**Pull request merged**